src/test/java/com/yelp/nrtsearch/server/grpc/UpdateDocValuesTest.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2025 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.grpc;
+
+import static org.junit.Assert.assertEquals;
+
+import com.yelp.nrtsearch.server.config.IndexStartConfig.IndexDataLocationType;
+import com.yelp.nrtsearch.server.grpc.AddDocumentRequest.MultiValuedField;
+import com.yelp.nrtsearch.server.grpc.SearchResponse.Hit;
+import io.grpc.StatusRuntimeException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class UpdateDocValuesTest {
+
+  @Rule public final TemporaryFolder folder = new TemporaryFolder();
+
+  @After
+  public void cleanup() {
+    TestServer.cleanupAll();
+  }
+
+  TestServer primaryServer;
+
+  String testPartialUpdateIndex = "test_partial_update_index";
+
+  private final List<Field> fields =
+      List.of(
+          Field.newBuilder()
+              .setName("primary_key")
+              .setType(FieldType._ID)
+              .setStoreDocValues(true)
+              .setSearch(true)
+              .build(),
+          Field.newBuilder()
+              .setName("update_field_one")
+              .setStoreDocValues(true)
+              .setType(FieldType.INT)
+              .build(),
+          Field.newBuilder()
+              .setName("update_field_two")
+              .setStoreDocValues(true)
+              .setType(FieldType.INT)
+              .build(),
+          Field.newBuilder()
+              .setName("non_updatable_field")
+              .setStoreDocValues(true)
+              .setType(FieldType.INT)
+              .setSearch(true)
+              .build());
+
+  @Test
+  public void testupdateDocValuesOnly() throws Exception {
+
+    primaryServer =
+        TestServer.builder(folder)
+            .withAutoStartConfig(true, Mode.PRIMARY, 0, IndexDataLocationType.LOCAL)
+            .build();
+
+    primaryServer.createIndex(testPartialUpdateIndex);
+    primaryServer.registerFields(testPartialUpdateIndex, fields);
+    primaryServer.startPrimaryIndex(testPartialUpdateIndex, -1, null);
+
+    primaryServer.addDocs(buildAddDocRequest(3).stream());
+    primaryServer.commit(testPartialUpdateIndex);
+    Thread.sleep(1000);
```

Comment:
Are these sleeps necessary. If you are waiting for an index refresh, it can be triggered on `primaryServer`